### PR TITLE
Create a control unittesting project

### PR DESCRIPTION
**Pull request by zadjii-msft**
Does what it says on the can.

This is a follow up to #9472. Now that we have a control .lib, we can add tests for it. 

Unfortunately, the `TermControl` itself is a horrible mess. So this new unittest lib is empty for now. I'm working on actual tests as a part of #6842, but this PR is here to keep the diffs smaller.

Also, apparently `server.vcxproj` had the wrong GUID in it.

* [x] I work here
* [x] Adds tests

### CODE REVIEW

File: src/server/lib/server.vcxproj [hunk 1]
```diff
@@ -59,7 +59,7 @@
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\..\host\proxy\Host.Proxy.vcxproj">
-      <Project>{e437b604-3e98-4f40-a927-e173e818ea4b}</Project>
```
Comment (miniksa) [inline src/server/lib/server.vcxproj:62]: wait what?

**Review by miniksa — CHANGES_REQUESTED** [review]
Why'd that one guid change? leftover mistake that VS fixed for us?

File: src/server/lib/server.vcxproj (hunk 1 above)
Comment (zadjii-msft) [reply src/server/lib/server.vcxproj:62]: IDK man. I saved the sln in VS, and it tore `e437b604-3e98-4f40-a927-e173e818ea4b` out of the .sln. I did a Find All, and this was the only reference to that guid, and _this isn't even the GUID of `Host.Proxy`_

**Review by carlos-zamora — APPROVED** [review]
Michael's handling the guid change. So I'mma just approve this since I don't see anything else worth blocking over.

File: src/server/lib/server.vcxproj (hunk 1 above)
Comment (miniksa) [reply src/server/lib/server.vcxproj:62]: Alright fine.

**Review by miniksa — APPROVED** [review]

**Pull request merged**